include/vast/Util/Pipeline.hpp

```diff
@@ -65,9 +66,77 @@ namespace vast {
 
         virtual void schedule(pipeline_step_ptr step) = 0;
 
+        void print_on_error(llvm::raw_ostream &os) {
+            enableIRPrinting(
+                [](auto *, auto *) { return false; }, // before
+                [](auto *, auto *) { return true; },  // after
+                false,                                // module scope
+                false,                                // after change
+                true,                                 // after failure
+                os
+            );
+        }
+
         llvm::DenseSet< pass_id_t > seen;
     };
 
+    struct with_snapshots : mlir::PassInstrumentation
```

Comment:
Add documentation to `docs/Tools/vast-front.md`

Reply:
Also might be worth to mention in `docs/GettingStarted/debug.md` in `### Pipeline Information`.